src/components/CourseCard/EmailSettingsModal.jsx

```diff
@@ -1,57 +1,138 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import {Button, CheckBox, Icon, Modal} from '@edx/paragon';
+import { Input, Modal, StatusAlert } from '@edx/paragon';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 import './EmailSettingsModal.scss';
 
+class EmailSettingsModal extends Component {
+  state = {
+    hasEmailsEnabled: this.props.hasEmailsEnabled || false,
+    isSubmitting: false,
+    isFormChanged: false,
+    error: null,
+  };
 
-const EmailSettingsModal = (props) => {
-  const modalRef = React.createRef();
-  const {
-    title, submitting, onClose, onEmailSettingsChange,
-  } = props;
-
-  return (
-    <React.Fragment>
-      <Modal
-        ref={modalRef}
-        title={title}
-        body={
-          <React.Fragment>
-            <CheckBox
-              name="email_settings"
-              label={
-                <React.Fragment>
-                  <span>
-                    Receive course emails such as reminders, schedule updates,
-                    and other critical announcements.
-                  </span>
-                  &nbsp; &nbsp;
-                  {submitting && <Icon className={['fa', 'fa-spinner', 'fa-spin', 'mr-2']} />}
-                </React.Fragment>
-              }
-              disabled={submitting}
-              onChange={onEmailSettingsChange}
-            />
-          </React.Fragment>
-        }
-        onClose={onClose}
-        closeText={submitting ? <Icon className={['fa', 'fa-spinner', 'fa-spin', 'mr-2']} /> : 'Save Settings'}
-        open
-      />
-    </React.Fragment>
-  );
-};
+  handleEmailSettingsChange = (e) => {
+    const { hasEmailsEnabled } = this.props;
+    const isChecked = e.target.checked;
+    this.setState({
+      isFormChanged: isChecked !== hasEmailsEnabled,
+      hasEmailsEnabled: isChecked,
+    });
+  };
 
-EmailSettingsModal.defaultProps = {
-  submitting: false,
-};
+  handleSaveButtonClick = () => {
+    const { hasEmailsEnabled } = this.state;
+    // TODO: Make API POST request to set the new email settings.
+    this.setState({
+      isSubmitting: true,
+    }, () => {
+      /* This callback function simulates an API call to save the
+       * new email settings. There are lines below to either
+       * resolve (200 response) or reject the promise (error response). To
+       * see what happens when the promise is rejected, switch the comments.
+       */
+
+      // eslint-disable-next-line no-unused-vars
+      new Promise((resolve, reject) => {
+        setTimeout(resolve, 2000);
+        // setTimeout(() => reject(new Error('Network Error')), 2000);
+      })
+        .then(() => {
+          this.props.onClose(hasEmailsEnabled);
+        })
+        .catch((error) => {
+          this.setState({
```

Comment:
If there's an error, update the `EmailSettingsModal` state to show the error `StatusAlert`.